ddtrace/internal/native/__init__.py

```diff
@@ -1 +1,42 @@
+import os
+import sys
+from typing import Dict
+
 from ._native import DDSketch  # noqa: F401
+from ._native import PyConfigurator
+
+
+def get_configuration_from_disk(debug_logs: bool = False, file_override="") -> Dict[str, str]:
+    """
+    Retrieves the tracer configuration from disk. Calls the PyConfigurator object
+    to read the configuration from the disk using the libdatadog shared library
+    and returns the corresponding configuration
+    See https://github.com/DataDog/libdatadog/blob/06d2b6a19d7ec9f41b3bfd4ddf521585c55298f6/library-config/src/lib.rs
+    for more information on how the configuration is read from disk
+    """
+    configurator = PyConfigurator(debug_logs)
+    configurator.set_envp(["%s=%s" % (k, v) for k, v in os.environ.items()])
+    configurator.set_args(sys.argv)
+
+    # Set the file override if provided. Only used for testing purposes.
+    if file_override:
+        configurator.set_file_override(file_override)
+
+    return configurator.get_configuration()
+
+
+def _apply_configuration_from_disk():
+    """
+    Sets the configuration from disk as environment variables.
+    This is not ideal and we should consider a better mechanism to
+    apply this configuration to the tracer.
+
+    Currently here is the order of precedence (higher takes precedence):
+    1. Dynamic remote configuration
+    2. Runtime configuration (ie fields set manually by customers / from the ddtrace code)
+    3. Configuration from disk
+    4. Environment variables
+    5. Default values
+    """
+    for key, value in get_configuration_from_disk().items():
+        os.environ[key] = str(value).lower()
```

Comment:
not sure I love this 🤔 

I don't have a good suggestion, but modifying system state in this way just _feels_ weird.

would we expect that a user should be able to access/see the configuration applied by accessing it at `os.environ[key]` ?

Reply:
It does really feel weird to me too; and to be fair it's a hack to make it work quickly. There is no global configuration object that is used everywhere that we can use.

Hopefully in the not-so-far future we'll be able to properly apply this configuration and track its origin; but this is a whole other problem to tackle.

Reply:
Well; looks like something else changed because now it passes. I re-enabled it 😕